cmd/agentless-scanner/main.go

```diff
@@ -6,5 +6,305 @@
 // Package main implements the agentless-scanner command.
 package main
 
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/signal"
+	"path"
+	"syscall"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/agentless/runner"
+	"github.com/DataDog/datadog-agent/pkg/agentless/types"
+
+	// DataDog agent: config stuffs
+	commonpath "github.com/DataDog/datadog-agent/cmd/agent/common/path"
+	"github.com/DataDog/datadog-agent/comp/core"
+	compconfig "github.com/DataDog/datadog-agent/comp/core/config"
+	complog "github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/comp/core/log/logimpl"
+	"github.com/DataDog/datadog-agent/comp/core/secrets"
+	"github.com/DataDog/datadog-agent/comp/forwarder/eventplatform"
+	"github.com/DataDog/datadog-agent/comp/forwarder/eventplatform/eventplatformimpl"
+	"github.com/DataDog/datadog-agent/comp/forwarder/eventplatformreceiver/eventplatformreceiverimpl"
+	pkgconfig "github.com/DataDog/datadog-agent/pkg/config"
+	pkgconfigsetup "github.com/DataDog/datadog-agent/pkg/config/setup"
+	"github.com/DataDog/datadog-agent/pkg/pidfile"
+	"github.com/DataDog/datadog-agent/pkg/security/utils"
+	"github.com/DataDog/datadog-agent/pkg/util/flavor"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	ddogstatsd "github.com/DataDog/datadog-go/v5/statsd"
+
+	_ "modernc.org/sqlite" // sqlite driver, used by github.com/knqyf263/go-rpmdb
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	defaultWorkersCount = 15
+	defaultScannersMax  = 8 // max number of child-process scanners spawned by a worker in parallel
+)
+
+var statsd *ddogstatsd.Client
+var eventForwarder eventplatform.Component
+
+var globalFlags struct {
+	configFilePath string
+	diskMode       types.DiskMode
+	defaultActions []types.ScanAction
+	noForkScanners bool
+}
+
+func runWithModules(run func(cmd *cobra.Command, args []string) error) func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		return fxutil.OneShot(
+			func(_ complog.Component, _ compconfig.Component, evp eventplatform.Component) error {
+				eventForwarder = evp
+				return run(cmd, args)
+			},
+			fx.Supply(core.BundleParams{
+				ConfigParams: compconfig.NewAgentParams(globalFlags.configFilePath),
+				SecretParams: secrets.NewEnabledParams(),
+				LogParams:    logimpl.ForDaemon(runner.LoggerName, "log_file", pkgconfigsetup.DefaultAgentlessScannerLogFile),
+			}),
+			core.Bundle(),
+			eventplatformimpl.Module(),
+			fx.Supply(eventplatformimpl.NewDefaultParams()),
+			eventplatformreceiverimpl.Module(),
+		)
+	}
+}
+
```

Comment:
I have realised the linter [lint_fxutil_oneshot_test](https://github.com/DataDog/datadog-agent/pull/23097/files#diff-f048224c30106b98e0d1e2937484c4667dab647d7175f85407c2889746fed8e0) was disabled.

This linter is very important because it checks there is no missing dependencies. Without this test, if a new component is added to Config, Log or eventplatformreceiver then this command will fails **at runtime**. I can help writing a test for your `fxutil.OneShot`.

Reply:
You can find a good example how to do it in [this PR](https://github.com/DataDog/datadog-agent/pull/21461), especially by look at [here](https://github.com/DataDog/datadog-agent/pull/21461/files#diff-7a7312dc54006eb70febc3455c6e3b97454f762fc18e6731adbb5e97b412017aR165-R223) and [here](https://github.com/DataDog/datadog-agent/pull/21461/files#diff-5153cd2f64ecba7465d48152e7944f13f04cd5184b892ee7bb5978d60213d6e4R24-R31)